packages/xdl/src/Project.js

```diff
@@ -1865,34 +1865,179 @@ function getWebpackInstance() {
   return webpackDevServerInstance;
 }
 
-async function startWebpackServerAsync(projectRoot, options, verbose) {
-  if (webpackDevServerInstance) {
-    ProjectUtils.logError(projectRoot, 'expo', 'Webpack is already running.');
-    return;
+const appDirectory = fs.realpathSync(process.cwd());
+const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
+const paths = {
+  yarnLockFile: resolveApp('yarn.lock'),
+  appTsConfig: resolveApp('tsconfig.json'),
+  appPackageJson: resolveApp('package.json'),
+};
+
+const devSocket = {
+  warnings: warnings => ProjectUtils.logInfo(projectRoot, 'warnings', warnings),
+  errors: errors => ProjectUtils.logInfo(projectRoot, 'errors', errors),
+};
+const useYarn = fs.existsSync(paths.yarnLockFile);
+const appName = require(paths.appPackageJson).name;
+
+import address from 'address';
+import formatWebpackMessages from 'react-dev-utils/formatWebpackMessages';
+import clearConsole from 'react-dev-utils/clearConsole';
+import { prepareUrls } from 'react-dev-utils/WebpackDevServerUtils';
+
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+('use strict');
```

Comment:
Did these imports, copyright header and `'use strict'` directive in the middle of the file come from some accidental copy paste?

Reply:
oops, I'll move this to another file since it's getting kinda bulky.